### DISCUSSION TITLE
Feature request: Sentry Instrumentation for Dart io HttpClient.

### DISCUSSION THREAD

**Discussion by silverhairs** [getsentry/sentry-dart#1483] [Ideas]
Out of the box, when performing http request in things such as the `NetworkImage` widget or the `CachedNetworkImagePackage` flutter uses dart's io [HttpClient](https://api.flutter.dev/flutter/dart-io/HttpClient-class.html) class. Currently Sentry does not have a package to instrument that. It's easy to override the underlying `HttpClient` with a custom implementation by extending [HttpOverrides](https://api.flutter.dev/flutter/dart-io/HttpOverrides-class.html) and overriding the `createHttpClient` method; but we cannot pass [dio](https://pub.dev/packages/dio) and [http.Client](https://pub.dev/packages/http) which are the two http client packages that sentry have instrumentation for. I think it would be important for sentry to support the native classes before going into the 3rd party libraries, this will reduce dependency on external library, and as a developer, I will not be obliged to depend on a 3rd party if I just want to instrument network request.

**Comment by marandaneto** [comment]
Yes, relates to https://github.com/getsentry/sentry-dart/issues/539 which is a feature request already, thanks for raising this up again, maybe we should raise the priority.
Would you like to contribute with a PR?

**Reply by silverhairs** [reply]
@marandaneto sure, I'd love to
